add document about stale read transaction (#5809)
pingcap/docs: stale-read.md

---
title: Usage Scenarios of Stale Read
summary: Learn about Stale Read and its usage scenarios.
---

# Usage Scenarios of Stale Read

This document describes the usage scenarios of stale read. Stale read is a mechanism that TiDB applies to read historical versions of data stored in TiDB. With this mechanism, you can read the corresponding historical data of a specific point in time or within a specified time range, and thus save the latency brought by data replication between storage nodes.

> Internally, stale read allows TiDB to read from any replica the data of the specified point in time or the data as new as possible within the specified time range, and to always ensure the data consistency constraint during the reading process.

Reviewer: ```suggestion
In terms of the internal implementation, Stale Read allows TiDB to read from any replica the data of the specified point in time or the data as new as possible within the specified time range, and to always ensure the data consistency constraint during the reading process.
```

## Scenario examples

> + Scenario one: If a transaction only involves read operations and is tolerant of data staleness to some extent, you can use stale reads to get historical data. With stale read, by sacrificing some real-time performance, TiDB sends the query requests to any replica of the corresponding data and thus increases the throughput of query executions.

Reviewer: ```suggestion
+ Scenario one: If a transaction only involves read operations and is tolerant of data staleness to some extent, you can use Stale Read to get historical data. Using Stale Read, TiDB sends the query requests to any replica of the corresponding data at the expense of some real-time performance, and thus increases the throughput of query executions.
```

> + Scenario two: In some scenarios where small tables are queried, if strongly consistent reads are used, data may be concentrated on a certain storage node, causing the query pressure to be concentrated on that node as well. Therefore, the node may become a bottleneck for the whole query. With stale read, TiDB distributes the query requests to each replica of the corresponding data, which can improve the overall query throughput and significantly improve the query performance.

Reviewer: ```suggestion
+ Scenario two: In some scenarios where small tables are queried, if strongly consistent reads are used, data might be concentrated on a certain storage node, causing the query pressure to be concentrated on that node as well. Therefore, that node might become a bottleneck for the whole query. With Stale Read, TiDB distributes the query requests to each replica of the corresponding data, which can improve the overall query throughput and significantly improve the query performance.
```

> + Scenario three: In some scenarios of geo-distributed deployment, if strongly consistent follower reads are used, in order to make sure that the data read from the Follower is consistent with that stored in the Leader, TiDB requests `Readindex` from different data centers for verification, which increases the access latency for the whole query process. With stale read, by sacrificing some real-time performance, TiDB accesses the replica in the current data center to read the corresponding data, which avoids network latency brought by cross-center connection and reduces the access latency for the entire query. For more information, see [Local Read under Three Data Centers Deployment](/best-practices/three-dc-local-read.md).

Reviewer: ```suggestion
+ Scenario three: In some scenarios of geo-distributed deployment, if strongly consistent follower reads are used, to make sure that the data read from the Followers is consistent with that stored in the Leader, TiDB requests `Readindex` from different data centers for verification, which increases the access latency for the whole query process. With Stale Read, TiDB accesses the replica in the current data center to read the corresponding data at the expense of some real-time performance, which avoids network latency brought by cross-center connection and reduces the access latency for the entire query. For more information, see [Local Read under Three Data Centers Deployment](/best-practices/three-dc-local-read.md).
```
